### DISCUSSION TITLE
Is it possible to create an extension that adds snippet variables?

### DISCUSSION THREAD

**Discussion by maximedupre** [microsoft/vscode-discussions#1274] [Extension Development QnA] [answered]
There are some built-in variables such as `TM_FILENAME` and I would like to extend this list of default variables.

For example, I would like to add `TM_METHODNAME`. Obviously, this would be language-specific (e.g. the extension would be for TypeScript snippets).

The [Language Extensions](https://code.visualstudio.com/api/language-extensions/overview#declarative-language-features) page isn't very informative in that regard, as it mostly talks about the use case of bundling snippets.

I've also seen a couple of GH tickets discussing "custom variables", but they mostly confused me lol

Is it possible to extend the list of snippet variables with an extension?

**Comment by isidorn** [comment, marked as answer]
Yes. These docs should help https://code.visualstudio.com/api/language-extensions/snippet-guide
